monkey/monkey_island/cc/agent_event_handlers/scan_event_handler.py

```diff
@@ -56,11 +57,22 @@ def handle_tcp_scan_event(self, event: TCPScanEvent):
 
         try:
             target_machine = self._get_target_machine(event)
+            source_node = self._get_source_node(event)
 
             self._update_nodes(target_machine, event)
+            self._update_tcp_connections(source_node, target_machine, event)
         except (RetrievalError, StorageError, UnknownRecordError):
             logger.exception("Unable to process tcp scan data")
 
+    def _get_source_node(self, event: AbstractAgentEvent) -> Node:
+        machine = self._get_source_machine(event)
+        try:
+            return [
+                node for node in self._node_repository.get_nodes() if node.machine_id == machine.id
+            ][0]
+        except KeyError:
+            raise UnknownRecordError(f"Source node for event {event} does not exist")
```

Comment:
```suggestion
        except IndexError:
            raise UnknownRecordError(f"Source node for event {event} does not exist")
```
There's no test that exercises this, though.